samples/OriginalUriBaseIds.sarif

```diff
@@ -0,0 +1,77 @@
+{
+  "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json",
+  "version": "2.1.0",
+  "properties": {
+    "comment": "This sample demonstrates the use of originalUriBaseIds."
+  },
+  "runs": [
+    {
+      "tool": {
+        "driver": {
+          "name": "SarifSamples"
+        }
+      },
+      "originalUriBaseIds": {
+        "PROJECTROOT": {
+          "description": {
+            "text": "The root directory for all project files."
+          },
+          "properties": {
+            "comment": "The SARIF producer has chosen not to specify a URI for PROJECTROOT. See §3.14.14, NOTE 1, for an explanation."
+          }
+        },
+        "SRCROOT": {
+          "uri": "src",
+          "uriBaseId": "PROJECTROOT",
+          "description": {
+            "text": "The root of the source tree."
+          },
+          "properties": {
+            "comment": "SRCROOT is expressed relative to PROJECTROOT."
+          }
+        }
+      },
+      "results": [
+        {
+          "ruleId": "SMP0001",

```

Comment:
>SMP0001 [](start = 21, length = 7)

How is SMP pronounced? 'simp'? if so, unfortunate. Also 'too many zeros'. :) 

How about a convention of SAMPXXX for the work we do here? I know this is a bit of a departure from conventions used by tools that reasonable allocation a range of 9999 potential issues. I think the adjustment helps with readability here. i.e., I like 'SAMP001' better than 'SMP0001'. Thoughts? This is nearly entirely in the realm of opinion...

Reply:
I'm SMPathetic to your concern about SMP, but I don't want to model, in our entire corpus of test assets, a pattern that we don't encourage tool authors to follow. How about `TST0001`, which I think you'll agree is obviously pronounced "test"?

Whatever we decide to do, I'll do in a separate PR that fixes all the files at once.

---
In reply to: [414792603](https://github.com/microsoft/sarif-tutorials/pull/14#discussion_r414792603) [](ancestors = 414792603)

Reply:
There's no rigid standard here. Honestly, your string of zeros compromises readability and if you are adhering to some standard tooling convention, we don't typically start indexing from 1, these lower-level ids are almost always reserved for internal error codes. CS1002 is more typical. 

How about TUT1001, TUT1002, etc. or TST1001, TST1002, etc.



---
In reply to: [414849147](https://github.com/microsoft/sarif-tutorials/pull/14#discussion_r414849147) [](ancestors = 414849147,414792603)

Reply:
Sold!

---
In reply to: [414852413](https://github.com/microsoft/sarif-tutorials/pull/14#discussion_r414852413) [](ancestors = 414852413,414849147,414792603)